(FACT-1660) Add Solaris zone detector

caseywilliams opened the pull request:
Adds a detector that checks the `zonename` executable to identify
Solaris zones. The global zone is not considered a valid result, but for
non-global zones, a valid result contains metadata reporting the zone
name, id, uuid, brand, and IP type; These are all discovered via `zoneadm`.

Example output:
```
  zone => {
    brand => "solaris",
    id => 1,
    ip_type => "excl",
    name => "testzone",
    uuid => "451df271-059c-ec89-c542-e36bbc5b0293"
  }
```

Like the LDom detector (#25), this will need an agent PR to take effect, since we're not building whereami on solaris yet. This detector did work for me in a test build, though.

caseywilliams committed:
(FACT-1660) Detect Solaris zones

Adds a detector that checks the `zonename` executable to identify
Solaris zones. The global zone is not considered a valid result, but for
non-global zones, a valid result contains metadata reporting the zone
name, id, uuid, brand, and IP type; These are all discovered via
`zoneadm`.

branan commented:
As for the LDOM detector, would be good to have fixtures and tests for the non-virtual case.

caseywilliams committed:
(maint) Add missing AIX lpar tests to CMakeLists

caseywilliams commented:
Rebased and updated with a test case for a global zone with several other configured zones. If no zone is found, the lack of `zonename` output will preclude any attempts to fill in metadata via `zoneadm`, but as in the LDom PR, there isn't much of a way to test that as things stand.

I also snuck in a maint commit that includes some tests for the AIX lpar and wpar detectors, which I forgot to include in CMakeLists.txt in my last PR 😓 . I can definitely split that into another PR if necessary.
